Projekt Bericht (#245)
Schmiddl99/experimenteverwaltung-i2: Dokumentationen/Projekt Bericht/Durchführung.adoc

=== Iteration 08 (02.06.2021-16.06.2021)

*Iterations Ziele:*

* Implementierung von TC03
* Implementierung von UC02, UC08, UC09
* Erstellung von Anwender und Admin Test
* Inkrement bei TSM5 abhalten 

==== Implementierung von TC03

Der TC03 wurde erfogreich implementiert und durchgeführt.

==== Implemtentierung der Use Cases

Bei der Implementierung wurde wieder Pairprogramming angewendet. 

==== Erstellung von Anwender und Admin Test

Um nochmals die Funktionalität und auch das intuitive Verständnis unseres System zu überprüfen zu können, erstellten wir geführte Anwender und Admin Testaufgaben. Der Anwender hatte die Aufgabe eine Vorlesung anzulegen und davon eine Wochenplan-PDF zu erstellen. Der Admin hatte die Aufgabe sich den Wochenplan des Anwenders anzusehen und auszudrucken. Die Aufgaben waren in mehere Unteraufgaebn unterteilt und enthielten mit fortschreitender Aufgabenstellung immer weniger detailierte Beschreibungen der auszuführenden Aufgaben. Hintergrund hier war, dass wir testen wollten, ob unsere Funktionen des Systems auch verstanden werden, wenn es wenig Hilfestellung dazu gibt. 

==== Inkrement bei TSM5 abhalten (10.06.2021)

> Am Anfang wurde das System vom Projektmanager vorgestellt. Anwesend waren hier Frau Dr. Krawietz, Herr Heisig und Herr Zirkelbach (Coach). Die Vorstellung verlief sehr gut bis auf einen Fehler, dass bei nicht ausreichender Bildschirmbreite sich die Buttons der Menüleiste übereinandergeschoben haben und somit das Menü Inhalte verdeckt hat.

Reviewer: ```suggestion
Am Anfang wurde das System vom Projektmanager vorgestellt. Anwesend waren hier Frau Dr. Krawietz, Herr Heisig und Herr Zirkelbach (Coach). Die Vorstellung verlief sehr gut, bis auf einen Fehler, dass sich die Buttons der Menüleiste übereinandergeschoben haben und somit die Menüinhalte verdeckt wurden, bei nicht ausreichender Bildschirmbreite.
```

> Anschließend wurde vom Projektmanger der Vorschlag unterbreitet, dass wenn man auf die Umsetzung von UC10 verzichten würde das Team mehr Zeit für andere Anforderungen hätte. Die Lösung wäre hier, dass sie Studiengänge hart ins System gecodet werden würden und wenn es Änderungen bei den Studiengängen geben sollte, die nach Meinung des Teams selten seien, diese auch wieder im System zu ändern. Eine ensprechende Beschreibung und Anleitung würde in der Betriebsdokumentaion beigelegt.

Reviewer: ```suggestion
Anschließend wurde vom Projektmanger der Vorschlag unterbreitet, dass durch den Verzicht auf die Umsetzung von UC10 mehr Zeit für andere Anforderungen entstehen würde. Die Lösung wäre hier, dass die Studiengänge hart ins System gecodet werden und wenn es Änderungen bei den Studiengängen geben sollte, die nach Meinung des Teams selten seien, diese auch wieder im System geändert werden. Eine ensprechende Beschreibung und Anleitung sollte in der Betriebsdokumentation beigelegt werden.
```

> Ein weiterer Vorschlag war auch auf die Filterfunktion bei dem Journal zu verzichten, um auch hier mehr Zeit zu gewinnen. (Die Vorschläg waren im Vorfeld mit dem Team abgesprochen worden). Der Kunde stimmte bei beiden zu, bestand aber darauf, dass die Filterfunktion in die Zukunftsaussicht mitaufgenommen werden würde. Der Projekmanager bestätigte dies.  

Reviewer: ```suggestion
Ein weiterer Vorschlag war auch, auf die Filterfunktion im Journal zu verzichten, um auch hier mehr Zeit zu gewinnen. (Die Vorschläg waren im Vorfeld mit dem Team abgesprochen worden). Der Kunde stimmte bei beiden zu, bestand aber darauf, dass die Filterfunktion in die Zukunftsaussicht mitaufgenommen werden würde. Der Projekmanager bestätigte dies.  
```

Der Kunde wünschte nach der Vorstellung noch einige Änderungen: 

* Inscope Änderungen 
** Einpflegen der Studiengangsnamen 
** Bei der Veränderungen einer Vorlesung im Journal soll ein Button erscheinen mit "Experiment hinzufügen" damit es intuitiver wird  
** Bei der Dozentenwoche soll ein Button erscheinen mit "Drucken" für die Druckfunktionalität
* Outscope Änderungen 

> ** Es soll eine Notification geben die den Admin darauf hinweißt, dass an einem bereits ausgedruckten Wochenplan noch eine Änderung durch einen Dozenten geschen ist

Reviewer: ```suggestion
** Es soll eine Notification geben die den Admin darauf hinweist, dass an einem bereits ausgedruckten Wochenplan noch eine Änderung durch einen Dozenten geschehen ist
```

> ** Es soll möglich sein die Voreinstellung auch noch im nachhinein zu verändern. 

Reviewer: ```suggestion
** Es soll möglich sein, die Voreinstellung auch noch im Nachhinein zu verändern. 
```

> Die gewünschten Änderungen wurde alle akzeptiert, nur die Änderungen bezüglich der Notification löste beim Projektmanager Unverständnis aus, da genau dieser Vorschlag dem Kunden bei TSM2 gemacht wurde und abgelehnt wurde. 

Reviewer: ```suggestion
Die gewünschten Änderungen wurde alle akzeptiert, nur die Änderungen bezüglich der Notification löste beim Projektmanager Unverständnis aus, da genau dieser Vorschlag dem Kunden bei TSM2 gemacht wurde und auf Ablehnung traf.
```

Zum Abschluss des Meetings teilte der Kunde nochmals seine Annerkennung und Dankbarkeit lautstark mit dem Projektmanager. Der Kunde schien über das bisherige System sehr glücklich zu sein. 

> Aus Zeitgründen wurden die Anwender- und Admintests nicht im Meeting durchgeführt sondern, Frau Dr. Krawietz und Herr Heisig wurden gebeten diese Zuhause zu erledigen. Beide akzeptierten den Vorschlag und fanden die Idee mit den Usertests gut. 

Reviewer: ```suggestion
Aus Zeitgründen wurden die Anwender- und Admintests nicht im Meeting durchgeführt. Frau Dr. Krawietz und Herr Heisig wurden gebeten, diese Zuhause zu erledigen. Beide akzeptierten den Vorschlag und fanden die Idee mit den Usertests gut. 
```

> Auf Wunsch des Kunden wurde nach dem TSM5, dem Kunden und Frau Dr. Krawietz Zugang zu dem Testsystem gewährt. Einmal natürlich um die Usertests durchführen zu können, aber auch um das System besser kennenlernen zu können. Wir hielten dies für eine gute Idee, da wir glaubten es sei besser selbst mit dem System arbeiten zu können, als nur auf eine Präsentation angewiesen zu sein. Der Zugang wurde gewährt. 

Reviewer: ```suggestion
Auf Wunsch des Kunden wurde nach dem TSM5, dem Kunden und Frau Dr. Krawietz Zugang zu dem Testsystem gewährt. Einmal natürlich um die Usertests durchführen zu können, aber auch um das System besser kennenlernen zu können. Wir hielten dies für eine gute Idee, da wir glaubten, es sei besser selbst mit dem System arbeiten zu können, als nur auf eine Präsentation angewiesen zu sein. Der Zugang wurde gewährt. 
```

Zusätzlich konnte der Kunde sich so auch mit der Userverwaltung besser vertraut machen. 

==== Abschluss der Iteration 08

* Puls-Check I08 LINK !!!

> * Retrospective

Reviewer: ```suggestion
* Retrospektive
```

** Projektstatus: *grün*

> ** Das sehr positive Feedback des erste Inkrement mit dem Kunden freut uns sehr. Da wir im laufe des Projekt ja immer wieder Verständigungsschwierigkeiten hatten, war bei uns im das Risko präsent, dass der Kunde nicht zufrieden wäre mit unser Arbeit. Das dieser Fall nun nicht eingetreten war erleichterte uns sehr. 

Reviewer: ```suggestion
** Das sehr positive Feedback des erste Inkrement mit dem Kunden freut uns sehr. Da wir im Laufe des Projektes ja immer wieder Verständigungsschwierigkeiten hatten, war bei uns immer das Risko präsent, dass der Kunde nicht zufrieden wäre, mit unser Arbeit. Dass dieser Fall nun nicht eingetreten war, erleichterte uns sehr. 
```

* Lesson Learned

> ** Inkremente sind unglaublich wichtig, für den Kunden sowie für die Entwickler. Kein Wireframe oder Protokoll kann mit der Informationsdichte oder dem Verständnisaustausches einer wirklichen Vorführung mithalten. 

Reviewer: ```suggestion
** Inkremente sind unglaublich wichtig, für den Kunden sowie für die Entwickler. Kein Wireframe oder Protokoll kann mit der Informationsdichte oder dem Verständnisaustausch einer wirklichen Vorführung mithalten. 
```

=== Iteration 09 (16.06.2021-30.06.2021)

*Iterations Ziele:*

* Einfügen der gewünschten Buttons

> * Experimentefeld Geräte optimieren 

Reviewer: ```suggestion
* Experimentefeld "Geräte" optimieren 
```

* Projekt mit Kunde abschließen
* Auswertung der User Tests
* Installation des Systems
* Auslieferung des Systems

==== Einfügen der gewünschten Buttons

> Der Drucken Button bei der Wochenplanansicht und der Experiment Hinzufügen Button bei dem Journal wurden erfolgreich implementiert. 

Reviewer: ```suggestion
Der "Drucken" Button bei der Wochenplanansicht und der "Experiment Hinzufügen" Button bei dem Journal wurden erfolgreich implementiert. 
```

==== Experimentefeld optimieren 

> Da wir die Forderung nach einem Buchungssystem als abgeschlossen betrachteten, kamen wir der Aufgabe nach das Experimentenfeld Geräte zu optimieren. Das Ziel war es die Ausgabe der Geräte in der übersichtsliste zu verbessern, da der Kunde viel Zeit damit verschwendet die Geräte zu finden. 

Reviewer: ```suggestion
Da wir die Forderung nach einem Buchungssystem als abgeschlossen betrachteten, kamen wir der Aufgabe nach das Experimentenfeld "Geräte" zu optimieren. Das Ziel war es, die Ausgabe der Geräte in der Übersichtsliste zu verbessern, da der Kunde viel Zeit damit verschwendet, die Geräte zu finden. 
```

> Wir analysierten das Problem und kamen zu dem Schluss, dass hier eine Stringsuche am besten geeignet war. Hier war der Vorteil, dass die Geräte beliebig erweitert werden konnten ohne die Suche anpassen zu müssen. 

Reviewer: ```suggestion
Wir analysierten das Problem und kamen zu dem Schluss, dass hier eine Stringsuche am besten geeignet war. Hier war der Vorteil, dass die Geräte beliebig erweitert werden konnten, ohne die Suche anpassen zu müssen. 
```

==== TSM6 Projekt mit dem Kunden abschließen 

> Die Vorführung der Buttons und des optimierten Gerätefelds wurden von dem Kunden erfreut angenommen. Allerdings wurden von Ihm noch einige Änderungen gewünscht. Die meisten davon waren visuller Natur, wie die Umbennung von Buttons oder die Umfärbung von Buttons, aber eine war direkt so groß, dass der Projektmanager direkt im Meeting sie als Outscope deklarieren konnte. Der Kunde forderte noch die Einführung einer zusätzlichen Rolle, die des Assistenten. Eine genaue Auflistung der Forderungen findet man in der Vision !!!

Reviewer: ```suggestion
Die Vorführung der Buttons und des optimierten Gerätefelds wurden von dem Kunden erfreut angenommen. Allerdings wurden von Ihm noch einige Änderungen gewünscht. Die meisten davon waren visuller Natur, wie die Umbennung und Umfärbung von Buttons. Eine jedoch war direkt so groß, dass der Projektmanager sie direkt im Meeting als Outscope deklarieren konnte. Der Kunde forderte noch die Einführung einer zusätzlichen Rolle, die des Assistenten. Eine genaue Auflistung der Forderungen findet man in der Vision. LINK!!!
```

==== Update des Systementwurfs im Architecture-Notebook

> Durch die Änderungen des Kunden im TSM4 musste auch das Architecture-Notebook entsprechend angepasst werden. Die Änderungen wurden unter Aufsicht des Analysten vorgenommen, da er denn besten Überblick hatte. 

Reviewer 2: ```suggestion
Durch die Änderungen des Kunden im TSM4 musste auch das Architecture-Notebook entsprechend angepasst werden. Die Änderungen wurden unter Aufsicht des Analysten vorgenommen, da er den besten Überblick über die neuen Anforderungen hatte. 
```

==== C4 Modellerweiterungen 

Das C4 Modell wurde um AW04 erweitert und der UC03 wurde mit aufgenommen. Die Arbeit an UC03 ist nicht abgeschlossen worden.  

==== Implementierung von UC01

> Der UC01 wurde erfolgreich implemntiert. Um bei der Implementierungen einen perfekten Fachaustausch zu gewährtleiten, verwenden wir hier und bei jeder weiteren Implementierung die Pairprogrammierung. An der Pairprogrammierung waren der Developer und der Analyst beteiligt. 

Reviewer 2: ```suggestion
Der UC01 wurde erfolgreich implementiert. Um bei der Implementierungen einen perfekten Fachaustausch zu gewährleisten, verwenden wir hier und bei jeder weiteren Implementierung die Pairprogrammierung. An der Pairprogrammierung waren der Developer und der Analyst beteiligt. 
```

==== Fortlaufende Korrektur des Belegs

> Wir kamen zu dem Entschluss, dass es eine Verbesserung des Organisationsaufwands wäre, wenn wir die Verbesserungen des Belegs nun nicht mehr durchgehend planen würden, sondern dazu übergehen, sie mit normalen Arbeiten am Beleg zu verbinden. Dies wurde auch in der Work-Item-List mit einem durchgehend Issue realisiert und es wurde ein Dokument in Microsoft Teams erstellt mit den noch zu erledigten Verbesserungen, sodass nichts vergessen wird.  

Reviewer 2: ```suggestion
Wir kamen zu dem Entschluss, dass es eine Verbesserung des Organisationsaufwands wäre, wenn wir die Verbesserungen des Belegs nun nicht mehr durchgehend planen würden, sondern dazu übergehen, sie mit normalen Arbeiten am Beleg zu verbinden. Dies wurde auch in der Work-Item-List mit einem durchgehenden Issue realisiert und es wurde ein Dokument in Microsoft Teams erstellt mit den noch zu erledigten Verbesserungen, sodass nichts vergessen wird.  
```

==== Abschluss des *LCA-Meilensteins* und Beginn des *IOC-Meilensteins*

Der *IOC-Meilenstein* konnte durch die Implemntierung von UC01 erfolgreich begonnen werden.
Der *LCA-Meilenstein* konnte nicht erfolgreich abgeschlossen werden, da das Ziel bezüglich der alten TCs nicht erreicht wurde. 

==== Abschluss der Iteration 06

===== Probleme
[%header, cols="2,1,3"]
|===
| Problem | Status | Notizen
| Stark erhöhter Zeitaufwand bei der Reparatur der alten Test Cases| Gelöst 
| Da unsere Vorgänger leider keine genau Aufführung Ihrer Test Cases erstellt haben, müssen wir uns nun langsam die alten Test Cases erarbeiten. Dieser Umstand des erhöhten Zeitaufwandes wurde als Problem bei der Kosten-Nutzen-Abschätzung erkannt und in der Risklist festgehalten. Nach der Mitigations-Strategie der Risk List wurde eine Notfalllösung gefunden: die Arbeit an den alten Test Cases wird nun erst einmal stillgelegt, bis eine Lösung dafür gefunden wurde (Coach und Projektvorgänger wurden um Hilfe gebeten). Es wird die Arbeit an unseren aktuellen Test Cases vorgezogen. 

> |Corona bedingte Zeitmanagementprobleme | Gelöst | Da das Corona Hygienekonzept zur erneuten Schließung von Kindergärten geführt hat, kann {PG}, der Vater von zwei Kindern ist, sich nicht in dem Umfang dem Projekt widmen, wie er es möchte. Dies führt zu dem Problem, dass er bei seinem Workload unterstützt werden muss, was ein großes Zeitproblem mit sich bringt. (Coach wurde über dieses "außergewöhnliche" Problem informiert). Das Problem wurde in der Risklsit vermerkt und als Lösung wurde beschlossen, dass wenn Zeitprobleme jeder Art auftauchen sollten, diese so schnell wie möglich der Gruppe mitzuteilen sind, dass so schneller gehandelt werden kann. 

Reviewer 2: ```suggestion
|Corona bedingte Zeitmanagementprobleme | Gelöst | Da das Corona Hygienekonzept zur erneuten Schließung von Kindergärten geführt hat, kann {PG}, der Vater von zwei Kindern ist, sich nicht in dem Umfang dem Projekt widmen, wie er es möchte. Dies führt zu dem Problem, dass er bei seinem Workload unterstützt werden muss, was ein großes Zeitproblem mit sich bringt. (Coach wurde über dieses "außergewöhnliche" Problem informiert). Das Problem wurde in der Risklist vermerkt und als Lösung wurde beschlossen, dass wenn Zeitprobleme jeder Art auftauchen sollten, diese so schnell wie möglich der Gruppe mitzuteilen sind, sodass schneller gehandelt werden kann. 
```

|===

* Retrospektive
** Projektstatus: *gelb*
* Unsere Arbeitsschätzungen betrachten wir als GUT, obwohl nicht alle Iterationsziele erreicht werden konnten. Dies begründen wir mit dem oben beschriebenen Corona-Umständen. Dennoch werden wir nun unserer Planung für die nächste Iteration überdenken und die Arbeitskräfte besser verteilen.  

> * Das Gruppen- und Arbeitsklima ist im grünen Bereich. Siehe <<Iteration 06, Iterations-Umfrage>> 

Reviewer 2: Der Link hier führt ins leere oder?

=== Iteration 07 (19.05.2021-02.06.2021)

*Iterations Ziele:*

* Zusammenführen des UC05, UC06, UC07 in den UC03

> * Reperatur oder Entfernung alter TCs

Reviewer 2: ```suggestion
* Reparatur oder Entfernung alter TCs
```

* UC03 im C4 Modell abschließen 
* TCs für UC01 werden umgesetzt und formuliert  
* UC03 Implementierung  
* Erstellung von UC10 bezüglich AW01

==== Reperatur oder Entfernung alter TCs

Dieses Ziel konnte dirket am Anfang der Iteration erreicht werden, da {rb} den Tester {pg} unterstütze. Durch die Expertise im Bereich Codeding und Ruby on Rails von {rb} konnten die alten TCs validert werden. Somit betrachten wir hier nun die Elobration Phase als abgeschlossen und haben den *LCA-Meilenstein* erreicht.

Unsere Entscheidung, die alten TCs erst einmal ruhen zu lassen, die wir in der letzten Iteration getroffen hatten, wurde vom Coach positiv bestätigt. 

> Ein weiterer Teil unsere Notfallmanagements war Vincent Thelang zu kontaktiern, ob er uns vielleicht weiterhelfen könne. Obwohl wir seine Hilfe bezüglich der alten TCs zum Meetingzeitpunkt (01.06.2021) glücklicherweise nicht mehr benötigten, hielten wir dennoch einen Fachaustausch ab. Hier ergaben sich zwar keine neuen Ergebnisse, aber Herr Thelang meinte, dass wir auf einem guten Weg wären. An dem Fachaustausch beteiligten sich der ProjekTmanager und der Developer, da hier die Bezugsgößen von Projektorganisation und Projektumsetzung am größten waren. 

Reviewer 2: ```suggestion
Ein weiterer Teil unsere Notfallmanagements war Vincent Thelang zu kontaktieren, ob er uns vielleicht weiterhelfen könne. Obwohl wir seine Hilfe bezüglich der alten TCs zum Meetingzeitpunkt (01.06.2021) glücklicherweise nicht mehr benötigten, hielten wir dennoch einen Fachaustausch ab. Hier ergaben sich zwar keine neuen Ergebnisse, aber Herr Thelang meinte, dass wir auf einem guten Weg wären. An dem Fachaustausch beteiligten sich der Projektmanager und der Developer, da hier die Bezugsgrößen von Projektorganisation und Projektumsetzung am größten waren. 
```

Die alten TCs werden im Beleg dokumentiert, damit zukünftige Entwickler einen besseren Überblick über alle TCs haben und wir beschreiben können, was wir repariert haben oder warum wir TCs entfernt haben.  

==== Zusammenführen der UCs

> Bei der Betrachung der Use-Cases UC05, UC06 und UC07 bemerkten wir, dass wir bei Ihrer Erstellung zu granular vorgegenagen waren. Da sich alle Use-Cases auf die Dozentenwoche bezogen, überführten wir diese in UC03. Somit war ein leichteres Verständnis und größere Übersichtlichkeit gegeben. Hier beteiligte sich der Analyst tatkräftig, dass keine Anforderungen verloren gingen. 

Reviewer 2: ```suggestion
Bei der Betrachtung der Use-Cases UC05, UC06 und UC07 bemerkten wir, dass wir bei Ihrer Erstellung zu granular vorgegangen waren. Da sich alle Use-Cases auf die Dozentenwoche bezogen, überführten wir diese in UC03. Somit war ein leichteres Verständnis und größere Übersichtlichkeit gegeben. Hier beteiligte sich der Analyst tatkräftig, sodass keine Anforderungen verloren gingen. 
```

==== UC03 im C4 Modell abschließen

> Der UC03 wurde im C4 Modell erfolgreich umgestzt.

Reviewer 2: ```suggestion
Der UC03 wurde im C4 Modell erfolgreich umgesetzt.
```

==== TCs für UC01 werden umgesetzt und formuliert  
 

>  Die TCs für UC01 wurden erfolgreich implementiert. Das Resultat war erfolgreich.

Reviewer 2: ```suggestion
Die TCs für UC01 wurden erfolgreich implementiert. Das Resultat war erfolgreich.
```

==== UC03 Implementierung

> Der UC03 wurde Implementiert. Hier haben Developer und Analyst Pairprogrammierung betrieben. 

Reviewer 2: ```suggestion
Der UC03 wurde implementiert. Hier haben Developer und Analyst Pairprogrammierung betrieben. 
```

==== Erstellung für UC10 bezüglich AW01

Der UC10 wurde erstellt um die Studiengangsverwaltung zu ermöglichen.

==== Abschluss der Iteration 07

* Puls-Check I07 (siehe Puls Check im QM)
* Retrospektive
** Projektstatus: *grün*

> ** Das Feedback des Coaches und des Fachaustausch wurden sehr postiv vom Team aufgenommen. Wir fühlen in unserem Tun mehr bestärkt und stehen sicherer hinter unseren Entscheidungen.

Reviewer 2: ```suggestion
** Das Feedback des Coaches und des Fachaustausch wurden sehr positiv vom Team aufgenommen. Wir fühlen uns in unserem Tun mehr bestärkt und stehen sicherer hinter unseren Entscheidungen.
```

** Durch das gute Voranschreiten im Entwicklungsprozess, sind wir der Meinung die Hauptaufgabe des Buchungssystems erfolgreich abschließen zu können. 
** Der beabsichtigte Zeitaufwand konnte erfüllt werden.
** Das Gruppen- und Arbeitsklima ist im grünen Bereich.

=== Iteration 08 (02.06.2021-16.06.2021)

*Iterationsziele:*

> * Implementierung von TC03

Reviewer 2: Sicher, dass hier TC03 (Leerer Warenkorb darf nicht zur Buchung führen) gemeint ist?

Ein weiterer Vorschlag war auch, auf die Filterfunktion im Journal zu verzichten, um auch hier mehr Zeit zu gewinnen. (Die Vorschläg waren im Vorfeld mit dem Team abgesprochen worden). Der Kunde stimmte bei beiden zu, bestand aber darauf, dass die Filterfunktion in die Zukunftsaussicht mitaufgenommen werden würde. Der Projekmanager bestätigte dies.   
Der Kunde wünschte nach der Vorstellung noch einige Änderungen: 

* Inscope Änderungen 
** Einpflegen der Studiengangsnamen 
** Bei der Veränderungen einer Vorlesung im Journal soll ein Button erscheinen mit "Experiment hinzufügen" damit es intuitiver wird  
** Bei der Dozentenwoche soll ein Button erscheinen mit "Drucken" für die Druckfunktionalität
* Outscope Änderungen 

> ** Es soll eine Notification geben die den Admin darauf hinweist, dass an einem bereits ausgedruckten Wochenplan noch eine Änderung durch einen Dozenten geschehen ist

Reviewer 2: ```suggestion
** Es soll eine Benachrichtigung geben, die den Admin darauf hinweist, dass an einem bereits ausgedruckten Wochenplan noch eine Änderung durch einen Dozenten geschehen ist.
```

** Es soll möglich sein, die Voreinstellung auch noch im Nachhinein zu verändern.

> Die gewünschten Änderungen wurde alle akzeptiert, nur die Änderungen bezüglich der Notification löste beim Projektmanager Unverständnis aus, da genau dieser Vorschlag dem Kunden bei TSM2 gemacht wurde und auf Ablehnung traf. 

Reviewer 2: ```suggestion
Die gewünschten Änderungen wurde alle akzeptiert, nur die Änderungen bezüglich der Benachrichtigung löste beim Projektmanager Unverständnis aus, da genau dieser Vorschlag dem Kunden bei TSM2 gemacht wurde und auf Ablehnung traf. 
```

> Zum Abschluss des Meetings teilte der Kunde nochmals seine Annerkennung und Dankbarkeit lautstark mit dem Projektmanager. Der Kunde schien über das bisherige System sehr glücklich zu sein. 

Reviewer 2: ```suggestion
Zum Abschluss des Meetings teilte der Kunde nochmals seine Anerkennung und Dankbarkeit lautstark mit dem Projektmanager. Der Kunde schien über das bisherige System sehr glücklich zu sein. 
```

Zusätzlich konnte der Kunde sich so auch mit der Userverwaltung besser vertraut machen. 

==== Abschluss der Iteration 08

* Puls-Check I08 (siehe Puls Check im QM)
* Retrospektive
** Projektstatus: *grün*
** Das sehr positive Feedback des erste Inkrement mit dem Kunden freut uns sehr. Da wir im Laufe des Projektes ja immer wieder Verständigungsschwierigkeiten hatten, war bei uns immer das Risko präsent, dass der Kunde nicht zufrieden wäre, mit unser Arbeit. Dass dieser Fall nun nicht eingetreten war, erleichterte uns sehr.  
* Lesson Learned
** Inkremente sind unglaublich wichtig, für den Kunden sowie für die Entwickler. Kein Wireframe oder Protokoll kann mit der Informationsdichte oder dem Verständnisaustausch einer wirklichen Vorführung mithalten.  

=== Iteration 09 (16.06.2021-30.06.2021)

*Iterations Ziele:*

* Einfügen der gewünschten Buttons
* Experimentefeld "Geräte" optimieren 
* Projekt mit Kunde abschließen
* Auswertung der User Tests
* Installation des Systems
* Auslieferung des Systems

==== Einfügen der gewünschten Buttons

> Der "Drucken" Button bei der Wochenplanansicht und der "Experiment Hinzufügen" Button bei dem Journal wurden erfolgreich implementiert. 

Reviewer 2: ```suggestion
Der "Drucken" Button bei der Wochenplanansicht und der "Experiment Hinzufügen" Button beim Bearbeiten einer Vorlesung wurden erfolgreich implementiert. 
```

==== Experimentefeld optimieren 

Da wir die Forderung nach einem Buchungssystem als abgeschlossen betrachteten, kamen wir der Aufgabe nach das Experimentenfeld Geräte zu optimieren. Das Ziel war es die Ausgabe der Geräte in der Übersichtsliste zu verbessern, da der Kunde viel Zeit damit verschwendet die Geräte zu finden. 

Wir analysierten das Problem und kamen zu dem Schluss, dass hier eine Stringsuche am besten geeignet war. Hier war der Vorteil, dass die Geräte beliebig erweitert werden konnten, ohne die Suche anpassen zu müssen. 

==== TSM6 Projekt mit dem Kunden abschließen 

> Die Vorführung der Buttons und des optimierten Gerätefelds wurden von dem Kunden erfreut angenommen. Allerdings wurden von Ihm noch einige Änderungen gewünscht. Die meisten davon waren visuller Natur, wie die Umbennung und Umfärbung von Buttons. Eine jedoch war direkt so groß, dass der Projektmanager sie direkt im Meeting als Outscope deklarieren konnte. Der Kunde forderte noch die Einführung einer zusätzlichen Rolle, die des Assistenten. Eine genaue Auflistung der Forderungen findet man in der Vision. 

Reviewer 2: ```suggestion
Die Vorführung der Buttons und des optimierten Gerätefelds wurden von dem Kunden erfreut angenommen. Allerdings wurden von Ihm noch einige Änderungen gewünscht. Die meisten davon waren visueller Natur, wie die Umbenennung und Umfärbung von Buttons. Eine jedoch war direkt so groß, dass der Projektmanager sie direkt im Meeting als out-of-Scope deklarieren konnte. Der Kunde forderte noch die Einführung einer zusätzlichen Rolle, die des Assistenten. Eine genaue Auflistung der Forderungen findet man in der Vision. 
```

> Auch wurde hier geich die Auslieferung und deren Ablauf mit dem Kunden besprochen. Der Kunde war mit den Abläufen einverstanden. Die Installation wurde auf den 28-06 und den 29.06.2021 festgelegt. Der Ausliefertermin sollte am 01.07.2021 stattfinden. 

Reviewer 2: ```suggestion
Auch wurde hier gleich die Auslieferung und deren Ablauf mit dem Kunden besprochen. Der Kunde war mit den Abläufen einverstanden. Die Installation wurde auf den 28.06. und den 29.06.2021 festgelegt. Der Ausliefertermin sollte am 01.07.2021 stattfinden. 
```

==== Entscheidung über Ablehnung des Änderungswunschs vom Kunden 

> Beim Treffen am 23.07.2021 teilte uns der Kunde mit, dass er zahlreiche Änderungen wünscht. So sollten mehrere Buttons umbenannt sowie farblich und in der Form verändert werden. Außerdem wünschte er sich eine zusätzliche Rolle und die Programmlogik sollte auch an einer Stelle geändert werden. Darüber hinaus zum Umfang dieser Änderungen waren Benennungswünsche widersprüchlich zur Funktionalität der Buttons. Nach längerer Beratung entschieden wir uns, diese Änderungswünsche abzulehnen und diese in der Zukunftsvision zu dokumentieren. Wir begründeten diese Entscheidung mit folgenden Argumenten:

Reviewer 2: ```suggestion
Beim Treffen am 23.06.2021 teilte uns der Kunde mit, dass er zahlreiche Änderungen wünscht. So sollten mehrere Buttons umbenannt sowie farblich und in der Form verändert werden. Außerdem wünschte er sich eine zusätzliche Rolle und die Programmlogik sollte auch an einer Stelle geändert werden. Darüber hinaus zum Umfang dieser Änderungen waren Benennungswünsche widersprüchlich zur Funktionalität der Buttons. Nach längerer Beratung entschieden wir uns, diese Änderungswünsche abzulehnen und diese in der Zukunftsvision zu dokumentieren. Wir begründeten diese Entscheidung mit folgenden Argumenten:
```

> * Installation auf das Produktivsystem war für den 28.7.2021 geplant und sollte nicht verschoben werden 

Reviewer 2: ```suggestion
* Installation auf das Produktivsystem war für den 28.06.2021 geplant und sollte nicht verschoben werden 
```

* somit noch 2 Werktage Zeit für Umsetzung der Wünsche
* Umsetzung hätte aufgrund ihrer Widersprüchlichkeit und Tiefe aber eine komplette Iteration benötigt 
* Dokumentation der Testcases und des Anwenderhandbuchs wären inkonsistent geworden da diese bereits erstellt wurden und aus zeitlichen Gründen nicht mehr geändert werden könnten

Zusammenfassend lässt sich also sagen das wir die Änderung ablehnten da ihre Umsetzung das brechen mit unseren Entwicklungsprinzipien und Best Practices bedingt hätte. Ein Umsetzen wäre nur im sprichwörtlichen Schnelldurchlauf möglich gewesen, was die Wahrscheinlichkeit von unentdeckten Fehlern im Code und am Produktivsystem erhöht hätte. Diese wären aber ab dem 02.07.2021 für einen längeren Zeitraum nicht behebbar gewesen, da dann die Betreung durch unser Team endet. Deshalb war es für uns wichtiger sorgfältig getesteten Code mit einer stimmigen Dokumentation abzugeben als auf die Wünsche einzugehen. 

> Alle vom Wunsch betroffenen Teile wurden dem Kunden vorher mehrmals gezeigt und von ihm bestätigt. So waren alle z.B. alle Bezeichnungen bereits in den vorgelegten Wireframes enthalten. Auch beim dem vom Kunden durchgeführten Usertests hätte er diese Wünsche äußern können. Das Aufkommen eines solchen Änderungswunsches 2 Werktage vor dem Aufspielen auf das Livesystem zeigt aber auch, dass es uns trotz intensiven Bemühungen nicht gelungen ist, dem Kunden unser Arbeitsweise so zu erläutern das er sie ausreichend verstanden hat. Über die gesamte Dauer des Projekts haben wir die Beobachtung gemacht, das der Kunde Änderungen in Bereichen wünscht die er bereits abgenommen hatte, die Priorisierung von Abschnitten anpassen möchte und Vorschlägen, Beratungen und Erklärungen von uns kaum Beachtung schenkt.

Reviewer 2: ```suggestion
Alle vom Wunsch betroffenen Teile wurden dem Kunden vorher mehrmals gezeigt und von ihm bestätigt. So waren z.B. alle Bezeichnungen bereits in den vorgelegten Wireframes enthalten. Auch bei den vom Kunden durchgeführten Usertests hätte er diese Wünsche äußern können. Das Aufkommen eines solchen Änderungswunsches 2 Werktage vor dem Aufspielen auf das Livesystem zeigt aber auch, dass es uns trotz intensiven Bemühungen nicht gelungen ist, dem Kunden unser Arbeitsweise so zu erläutern das er sie ausreichend verstanden hat. Über die gesamte Dauer des Projekts haben wir die Beobachtung gemacht, dass der Kunde Änderungen in Bereichen wünscht die er bereits abgenommen hatte, die Priorisierung von Abschnitten anpassen möchte und Vorschlägen, Beratungen und Erklärungen von uns kaum Beachtung schenkt.
```

==== Auswerten von User Tests 

> Die Auswerung hat ergeben, dass Frau Dr. Krawietz mit den von uns gestellten Aufgaben, sehr gut zurecht kam und Ihr auch keine Fehler aufgefallen sind. Wir sehen uns hier bestätigt, dass wir die geforderte inuative Arbeitsweise auf dem System erreichen konnten, da Frau Dr. Krawietz keine Hilfestellung hatte und auch keinen Zugriff auf eine Anwenderdokumentation. 

Reviewer 2: ```suggestion
Die Auswertung hat ergeben, dass Frau Dr. Krawietz mit den von uns gestellten Aufgaben, sehr gut zurecht kam und Ihr auch keine Fehler aufgefallen sind. Wir sehen uns hier bestätigt, dass wir die geforderte intuitive Arbeitsweise auf dem System erreichen konnten, da Frau Dr. Krawietz keine Hilfestellung hatte und auch keinen Zugriff auf eine Anwenderdokumentation. 
```

=== Systemoptimierung und Übergabe an den Kunden 

*Iterations Ziele:*

* Einfügen der gewünschten Buttons
* Experimentefeld "Geräte" optimieren 
* Bugfixing der Textfelder
* Projekt mit Kunde abschließen
* Auswertung der User Tests
* Installation des Systems
* Auslieferung des Systems

==== Einfügen der gewünschten Buttons

Der "Drucken" Button bei der Wochenplanansicht und der "Experiment Hinzufügen" Button beim Bearbeiten einer Vorlesung wurden erfolgreich implementiert. 

==== Experimentefeld optimieren 

Da wir die Forderung nach einem Buchungssystem als abgeschlossen betrachteten, kamen wir der Aufgabe nach das Experimentenfeld Geräte zu optimieren. Das Ziel war es die Ausgabe der Geräte in der Übersichtsliste zu verbessern, da der Kunde viel Zeit damit verschwendet die Geräte zu finden. 

Wir analysierten das Problem und kamen zu dem Schluss, dass hier eine Stringsuche am besten geeignet war. Hier war der Vorteil, dass die Geräte beliebig erweitert werden konnten, ohne die Suche anpassen zu müssen. 

==== Bugfixing der Textfelder 

> Die Problematik das Textfelder bei dem Experimentefeld Gefahren übereinander geschoben werden wurde behoben. Hiermit endete die Weiterentwicklung am System und wir konnten den *IOC-Meilenstein* abschließen. 

Reviewer 2: ```suggestion
Die Problematik, dass Textfelder bei dem Experimentefeld "Gefahren" übereinander geschoben werden, wurde behoben. Hiermit endete die Weiterentwicklung am System und wir konnten den *IOC-Meilenstein* abschließen. 
```

==== Auswerten von User Tests 

Die Auswertung hat ergeben, dass Frau Dr. Krawietz mit den von uns gestellten Aufgaben, sehr gut zurecht kam und Ihr auch keine Fehler aufgefallen sind. Wir sehen uns hier bestätigt, dass wir die geforderte intuitive Arbeitsweise auf dem System erreichen konnten, da Frau Dr. Krawietz keine Hilfestellung hatte und auch keinen Zugriff auf eine Anwenderdokumentation. 
Von Herr Heisig haben wie leider keine ausgefüllten Unterlagen erhalten, aber auch er betonte im Meeting, dass er mit den Aufgaben keine Probleme hatte. 

==== Installation des Systems

Die Installation des Systems wurde am 28.06.2021 von unserem Developer {rb} durchgeführt und konnte erfolgreich, ohne Probleme abgeschlossen werden.

==== Auslieferung des Systems

> Die Auslieferung fand am 01.07.2021 in der HTW Z254 auf dem persönlichen Rechner des Kunden statt. Das Abgabeprotokoll wurde im Vorfeld dem Kunden per Mail geschickt und so von Ihm akzeptiert. Die vorbereiteten Szenarien wurde alle vom Kunden persönlich durchgeführt und lief fehlerfrei durch. Es wurden keine Anmerkungen vom Kunden gemacht. Zusätzlich zu den Dokumentationen wurde dem Kunden ein Teil der Vision übergeben, dessen Inhalt sich um die Zukunftsanforderungen und die noch umzusetzenden Änderungen bezog. Hintergund hier war, dass wenn der Kunden das Projekt in Zukunft nochmals optimieren möchte, er hier eine Zusammenfassung seiner Wünsche zur Verfügung hätte. 

Reviewer 2: ```suggestion
Die Auslieferung fand am 01.07.2021 in der HTW Z254 auf dem persönlichen Rechner des Kunden statt. Das Abgabeprotokoll wurde im Vorfeld dem Kunden per Mail geschickt und so von Ihm akzeptiert. Die vorbereiteten Szenarien wurden alle vom Kunden persönlich durchgeführt und liefen fehlerfrei durch. Es wurden keine Anmerkungen vom Kunden gemacht. Zusätzlich zu den Dokumentationen wurde dem Kunden ein Teil der Vision übergeben, dessen Inhalt sich um die Zukunftsanforderungen und die noch umzusetzenden Änderungen bezog. Hintergrund hier war, dass wenn der Kunden das Projekt in Zukunft nochmals optimieren möchte, er hier eine Zusammenfassung seiner Wünsche zur Verfügung hätte. 
```

Der Projektmanager unterschrieb im Namen des ganzen Teams und der Kunde zeichnete das Abgabeprotokoll gegen. Somit war die Abgabe abgeschlossen und der *PR-Meilenstein* konnte erfolgreich beendet werden. Der Kunde erhielt eine Kopie des unterschriebenen Abgabeprotokolls per Mail. 

==== Abschluss der Iteration 09

* Puls-Check I09 (siehe Puls Check im QM)
* Retrospektive
** Projektstatus: *grün*

> ** Als das Projekt endgültig beim Kunden abgegeben war, waren wir alle sehr erleichert und glücklich. Ein großes Projekt mit sehr viel Nervenaufwand und Arbeitszeit ging nun zu Ende. Auch über das Angebot des Kunden, dass es vielleicht möglich sei nächstes Semester das Projekt als Studentischehilfskraft fortzuführen, sahen wir als große Bestätigung des Kunden unsere Arbeit gegenüber. 

Reviewer 2: ```suggestion
** Als das Projekt endgültig beim Kunden abgegeben war, waren wir alle sehr erleichtert und glücklich. Ein großes Projekt mit sehr viel Nervenaufwand und Arbeitszeit ging nun zu Ende. Auch über das Angebot des Kunden, dass es vielleicht möglich sei nächstes Semester das Projekt als studentische Hilfskraft fortzuführen, sahen wir als große Bestätigung des Kunden unsere Arbeit gegenüber. 
```

** Das Gruppen- und Arbeitsklima ist im grünen Bereich. 
* Lesson Learned

> ** Das Inkrement so spät durchzuführen hat sich als Fehler erwiesen. Wir waren uns zwar sicher, dass wir mit den Wireframes und der Vorführung die endgültige Zustimmung des Kunden erreicht hätte, dem war aber leider nicht so. Wir mussten leider feststellen, dass der Kunden als er sich mit der Sotware allein beschäftigte, im nochmals Änderungswünsche eingefallen sind. In dieser Situation wäre es sicher besser gewesen, mit dem Kunden viele kleine Inkremente durchzuführen. 

Reviewer 2: ```suggestion
** Das erste Inkrement so spät durchzuführen hat sich als Fehler erwiesen. Wir waren uns zwar sicher, dass wir mit den Wireframes und der Vorführung die endgültige Zustimmung des Kunden erreicht hätten, dem war aber leider nicht so. Wir mussten leider feststellen, dass dem Kunden, als er sich mit der Software allein beschäftigte, nochmals Änderungswünsche eingefallen sind. In dieser Situation wäre es sicher besser gewesen, mit dem Kunden viele kleine Inkremente durchzuführen. 
```

> ** Das erste Inkrement so spät durchzuführen hat sich als Fehler erwiesen. Wir waren uns zwar sicher, dass wir mit den Wireframes und der Vorführung die endgültige Zustimmung des Kunden erreicht hätten, dem war aber leider nicht so. Wir mussten leider feststellen, dass dem Kunden, als er sich mit der Software allein beschäftigte, nochmals Änderungswünsche eingefallen sind. In dieser Situation wäre es sicher besser gewesen, mit dem Kunden viele kleine Inkremente durchzuführen. 

Reviewer: ```suggestion
** Das erste Inkrement so spät durchzuführen hat sich als Fehler erwiesen. Wir waren uns zwar sicher, dass wir mit den Wireframes und der Vorführung die endgültige Zustimmung des Kunden erreicht hatten, dem war aber leider nicht so. Wir mussten leider feststellen, dass dem Kunden, als er sich mit der Software allein beschäftigte, nochmals Änderungswünsche eingefallen sind. In dieser Situation wäre es sicher besser gewesen, mit dem Kunden viele kleine Inkremente durchzuführen. 
```
